### PR TITLE
feat(@desktop/wallet): New design for Transaction History

### LINKED ISSUE #7213: Implement transaction view as per new design
#7073

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
feat(@desktop/wallet): New design for Transaction History
fixes #7213

StatusQ PR: https://github.com/status-im/StatusQ/pull/879

### What does the PR do

Updates activity history to look like design (see https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=6105%3A75943 ) https://github.com/status-im/status-desktop/issues/7215

### Affected areas

Wallet

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://user-images.githubusercontent.com/60327365/187955665-e8ceb93f-c469-4ada-a0e9-7e26e9dd8435.mov

### CODE REVIEW

File: ui/imports/shared/controls/TransactionDelegate.qml [hunk 1]
```diff
@@ -1,148 +1,94 @@
...
-                    assetIcon.source = Style.png("tokens/DEFAULT-TOKEN@3x")
+    components: [
+        ColumnLayout {
+            Row {
+                Layout.alignment: Qt.AlignRight
+                spacing: 4
+                StatusIcon {
+                    color: isIncoming ? Theme.palette.successColor1 : Theme.palette.dangerColor1
+                    icon: "arrow-up"
+                    rotation: isIncoming ? 135 : 45
```
Comment (caybro) [inline ui/imports/shared/controls/TransactionDelegate.qml:58]: We have distinct arrow icons for up/down

File: ui/imports/shared/views/HistoryView.qml [hunk 2]
```diff
@@ -71,50 +68,76 @@ Item {
...
+            }
         }
 
-        onCountChanged: {
-            if (loadMoreButton.loadedMore)
-                transactionListRoot.positionViewAtEnd();
+        Component {
+            id: transactionDelegate
+            TransactionDelegate {
+                width: parent.width
```
Comment (caybro) [inline ui/imports/shared/views/HistoryView.qml:101]: No `parent` here

File: ui/imports/shared/views/HistoryView.qml [hunk 3]
```diff
@@ -71,50 +68,76 @@ Item {
...
             }
         }
 
-        ScrollBar.vertical: ScrollBar {
-            id: scrollBar
+        Component {
+            id: dateHeader
+            StatusListItem {
+                property var modelData
+                width: parent.width
```
Comment (caybro) [inline ui/imports/shared/views/HistoryView.qml:89]: No `parent` around

Comment (caybro) [reply ui/imports/shared/views/HistoryView.qml:89]: You can use `ListView.view.width`

File: ui/imports/shared/views/HistoryView.qml [hunk 4]
```diff
@@ -71,50 +68,76 @@ Item {
...
-        id: loadMoreButton
-        text: qsTr("Load More")
-        // TODO: handle case when requested limit === transaction count -- there
-        // is currently no way to know that there are no more results
-        enabled: !loadingImg.active && RootStore.historyTransactions.hasMore
-        anchors.right: parent.right
-        anchors.bottom: parent.bottom
-        anchors.bottomMargin: Style.current.padding
-        property bool loadedMore: false
+        ScrollBar.vertical: ScrollBar {}
```
Comment (caybro) [inline ui/imports/shared/views/HistoryView.qml:120]: Make use of `StatusScrollBar`

File: ui/imports/utils/Utils.qml [hunk 5]
```diff
@@ -295,6 +295,15 @@ QtObject {
         return qsTr("%1D").arg(diffDay)
     }
 
+    function formatShortDate(value, isDDMMYYDateFormat) {
+        const formatDDMMYY = "d MMMM yyyy"
```
Comment (caybro) [inline ui/imports/utils/Utils.qml:299]: Uhm.... :) OK, adding to my TODO as well

**Review by alaibe — APPROVED** [review]

**Comment by anastasiyaig** [comment]
@Khushboo-dev-cpp 

1. transaction list is not updated in real time, i am forced to relogin
2. when sending a token, i am seeing `ETH sent` instead of token name
3. when i am sending on Goerli, Ropsten transactions are shown 

Please take a look

File: ui/imports/shared/controls/TransactionDelegate.qml (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/controls/TransactionDelegate.qml:58]: image
This is the what I need on this case!

File: ui/imports/shared/controls/TransactionDelegate.qml [hunk 6]
```diff
@@ -1,148 +1,94 @@
...
-        onExited: {
-            transactionListItem.isHovered = false
+StatusListItem {
+    id: root
+
+    property var modelData
+    property string symbol
+    property bool isIncoming
+    property int transferStatus
+    property string currentCurreny
```
[the remark this answers is not in the document]
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/controls/TransactionDelegate.qml:18]: done

File: ui/imports/utils/Utils.qml (hunk 5 above)
Comment (Khushboo-dev-cpp) [reply ui/imports/utils/Utils.qml:299]: @caybro have addressed all your comments :) Please review and approve when you can!

**Comment by Khushboo-dev-cpp** [comment]
> @Khushboo-dev-cpp
> 
> 1. transaction list is not updated in real time, i am forced to relogin
> 2. when sending a token, i am seeing `ETH sent` instead of token name
> 3. when i am sending on Goerli, Ropsten transactions are shown
> 
> Please take a look

Hi Nastya,

> 1. transaction list is not updated in real time, i am forced to relogin
Raised a new bug https://github.com/status-im/status-desktop/issues/7266
> 2. when sending a token, i am seeing `ETH sent` instead of token name
Fixed this!
> 3. when i am sending on Goerli, Ropsten transactions are shown
and I actually dont see a Goreli transaction after a restart also, have reported it in the same bug above.

**Comment by anastasiyaig** [comment]
https://github.com/status-im/status-desktop/issues/7278 i logged this one to track the problem of scrolling and Load More button

**Review by anastasiyaig — APPROVED** [review]

**Review by caybro — APPROVED** [review]
Looks good!

i18n issues will be solved separately

**Pull request merged**